src/ArduinoIoTCloud.cpp

```diff
@@ -28,20 +33,24 @@ const static int thingIdSlot                               = 12;
 static ConnectionManager *getTimeConnection = NULL;
 
 static unsigned long getTime() {
-  if (!getTimeConnection) return 0;
-  unsigned long time = getTimeConnection->getTime();
-  if (!NTPUtils::isTimeValid(time)) {
-    debugMessage("Bogus NTP time from API, fallback to UDP method", 0);
-    time = NTPUtils(getTimeConnection->getUDP()).getTime();
-  }
-  return time;
```

Comment:
I'm not really convinced by this one; I think the fallback should be handled transparently at the top layer, not by every ConnectionManager (this will make the porting to new transports easier). The removal of pure virtual getTime() later on makes sense but creates a bit of confusion IMO

Reply:
I made the change in order to manage the fallback mechanism also for the CONNECTION_STATE_GETTIME state in the derived ConnectionManager classes.

Reply:
Reverted to the old version. Moving the logic into a a new PR.